pkg/syncer/metadatasyncer.go

```diff
@@ -747,6 +800,17 @@ func csiPVDeleted(ctx context.Context, pv *v1.PersistentVolume, metadataSyncer *
 		}
 
 	} else {
+		var err error
+		var volumeHandle string
+		if metadataSyncer.configInfo.Cfg.FeatureStates.CSIMigration && pv.Spec.VsphereVolume != nil {
+			volumeHandle, err = GetVolumeID(ctx, pv)
```

Comment:
if PV is already deleted, then VMS - DeleteVolumeInfo may have removed mapping, and associated disk may already be removed.

Reply:
This is handled by checking the claimRef at the beginning of this function